### PR TITLE
Made test_parse.py pep8 compliant, and fixed most pylint messages

### PR DESCRIPTION
**Pull request by hiker**
(except missing docstrings).

At least a small start ;)

### CODE REVIEW

File: src/fparser/tests/test_parser.py [hunk 1]
```diff
@@ -71,9 +71,18 @@
...
+     AssignedGoto, Assign, Assignment, Asynchronous, Backspace, Bind, Call, \
+     CallProtoArgument, CallStatement, Character, Check, Close, Common, \
+     ComputedGoto, Contains, Continue, Cycle, Data, Deallocate, Depend, \
+     Dimension, Else, ElseIf, ElseWhere, Endfile, Entry, Enumerator, \
+     Equivalence, Exit, External, FinalBinding, Flush, Format, ForallStmt, \
+     FortranName, GenericBinding, Goto, Implicit, Import, Inquire, Integer, \
+     Intent, Intrinsic, ModuleProcedure, Nullify, Open, Optional, Parameter, \
+     Pointer, PointerAssignment, Pause, Print, Private, Protected, Public, \
+     Read, Return, Rewind, Save, Sequence, SpecificBinding, Stop, Target, \
+     Threadsafe, Use, Value, Volatile, Wait, WhereStmt, Write
```
Comment (p-vitt) [inline src/fparser/tests/test_parser.py]: Does it make sense to replace `from something import *` with `from something import yadda, blah, and_whatever` if `yadda`, `blah`, and `and_whatever` is equal to `*`? In this case I think deactivating the pylint message with a reasonable comment would be fine, too.

Comment (hiker) [reply src/fparser/tests/test_parser.py]: I got recently bitten by a kind of import * (Fortran, ... but still) , so I decided to go this way - even though believe me I found it highly annoying ;) For me the big advantage is that you can search for any symbol and see where it comes from. Alternative we could use
```
import fparser.block_statements
```
et. All in all, I admit I don't care, if the maintainers prefer import * that's fine with me, too.

Comment (p-vitt) [reply src/fparser/tests/test_parser.py]: That's why we are doing `use xxx, only: yyy` everywhere :) I was just wondering whether there are other reasons besides that.

Comment (hiker) [reply src/fparser/tests/test_parser.py]: *grin* except that using ```only``` confused PSyclone (ticket to be opened there as soon as I have a test case), so I had to remove the ```only```. And that module defined 'min' as a variable - believe me, I thought I was going crazy, esp  ;)
No, no other reason - except that I don't now exactly what the maintainers prefer, so I played it safe ;)

Comment (rupertford) [reply src/fparser/tests/test_parser.py]: I prefer explicit!

File: src/fparser/tests/test_parser.py [hunk 2]
```diff
@@ -83,115 +92,132 @@ def parse(cls, line, label='', isfree=True, isstrict=False):
...
     item = reader.next()
     if not cls.match(item.get_line()):
-        raise ValueError('%r does not match %s pattern'%(line, cls.__name__))
+        raise ValueError('%r does not match %s pattern' % (line, cls.__name__))
     stmt = cls(item, item)
     if stmt.isvalid:
         # Check that we can successfully parse the string representation
         # of the parsed object
-        r = str(stmt)
+        r = str(stmt)  # pylint: disable=invalid-name
```
Comment (rupertford) [inline src/fparser/tests/test_parser.py]: Sorry, I think we should conform to the pep8 variable name requirements. I'm happy for you to not do this in this issue but please remove the pylint directive.

File: src/fparser/tests/test_parser.py [hunk 3]
```diff
@@ -83,115 +92,132 @@ def parse(cls, line, label='', isfree=True, isstrict=False):
...
         if not isstrict:
-            r1 = parse(cls, r, isstrict=True)
+            r1 = parse(cls, r, isstrict=True) # pylint: disable=invalid-name
```
Comment (rupertford) [inline src/fparser/tests/test_parser.py]: Same as previous comment

**Comment by hiker** [comment]
Sorry rupert, I apparently did indeed forget to run pep8 again after I resolved the conflicts. Should all be done now.

**Comment by rupertford** [comment]
No need to apologise about the missing pep8 fixes, I do it all the time.
Changes have fixed the pep8 warnings and all tests still pass (with one xfail).
We're good to merge.

**Pull request merged**